tripal/includes/TripalEntityController.inc

```diff
@@ -645,8 +645,9 @@ class TripalEntityController extends EntityAPIController {
           $field_id = $field['id'];
 
           // Add this field to the entity with default value.
-          if (!isset($queried_entities[$id]->$field_name)) {
-            $queried_entities[$id]->$field_name = array();
+          if (!isset($queried_entities[$id]->{$field_name})) {
+            $queried_entities[$id]->{$field_name} = array();
+            $entity->{$field_name} = array();
```

Comment:
I'm a bit leery of this change. It's my understanding the if a field is attached it should be "all the way" attached and have the language, the delta the values.  I can see how this fixes the problem but I'm worried it may cause other sideffects.